client/src/bastionlab/polars/remote_polars.py

```diff
@@ -1327,3 +1396,84 @@ def __map(self: LDF, func, **kwargs) -> None:
 class RemoteLazyGroupBy(Generic[LDF]):
     _inner: pl.internals.lazyframe.groupby.LazyGroupBy[LDF]
     _meta: Metadata
+
+
+def train_test_split(
+    *arrays: List["RemoteArray"],
+    train_size: Optional[float] = None,
+    test_size: Optional[float] = 0.25,
+    shuffle: Optional[bool] = False,
+    random_state: Optional[int] = None,
+) -> List["RemoteArray"]:
+    """
+    Split RemoteDataFrames into train and test subsets.
+
+    Args:
+        train_size (Optional[float] = None):
+            It should be between 0.0 and 1.0 and represent the proportion of the dataset to include in the train split.
+            If None, the value is automatically set to the complement of the test size.
+        test_size (Optional[float] =0.25):
+            It should be between 0.0 and 1.0 and represent the proportion of the dataset to include in the test split.
+            If None, the value is set to the complement of the train size.
+            If train_size is also None, it will be set to 0.25.
+        shuffle (Optional[bool] = False):
+            Whether or not to shuffle the data before splitting.
+        random_state (Optional[int] = -1):
+            Controls the shuffling applied to the data before applying the split.
+            Pass an int for reproducible output across multiple function calls.
+    """
+
+    from .remote_polars import FetchableLazyFrame
+
+    if len(arrays) == 0:
+        raise ValueError("At least one RemoteDataFrame required as input")
+
+    _train_rdf: RemoteArray = arrays[0]
+
+    train_size = 1 - test_size if train_size is None else train_size
+    test_size = 1 - train_size if test_size is None else test_size
+
+    arrays: List[ReferenceRequest] = [
+        ReferenceRequest(identifier=rdf.identifier) for rdf in arrays
+    ]
+
+    res = _train_rdf._meta._polars_client.stub.Split(
+        SplitRequest(
+            arrays=arrays,
+            train_size=train_size,
+            test_size=test_size,
+            shuffle=shuffle,
+            random_state=random_state,
+        )
+    )
+    res = [
+        FetchableLazyFrame._from_reference(
+            _train_rdf._meta._polars_client, ref
+        ).to_array()
+        for ref in res.list
+    ]
+    return res
+
+
+class RemoteArray(RemoteLazyFrame):
```

Comment:
i still dont get what remote array is :sweat_smile: 

if i understand correctly, it's something that is client-only, for remotedataframe.to_array().to_tensor()? cant we just have remotedataframe.to_tensor?
i cant see any RemoteArray in the rust code

Reply:
You can ask Lucas about RemoteArray's.
He did the framing

Reply:
Okay thanks, i talked with Lucas

Reply:
Do remote lazy frames support .to_tensor directly ? (as sugar for .to_array().to_tensor())